packages/wp-source/src/libraries/handlers/__tests__/date.tests.ts

```diff
@@ -120,4 +120,34 @@ describe("date", () => {
     expect(api.get.mock.calls).toMatchSnapshot();
     expect(store.state.source).toMatchSnapshot();
   });
+
+  test("overwrites the data when fetched with { force: true }", async () => {
+    api.get = jest
+      .fn()
+      .mockResolvedValueOnce(
+        mockResponse(date2019Posts, {
+          "X-WP-Total": "3",
+          "X-WP-TotalPages": "1",
+        })
+      )
+      .mockResolvedValueOnce(
+        mockResponse(date2019PostsPage2, {
+          "X-WP-Total": "3",
+          "X-WP-TotalPages": "1",
+        })
+      );
+
+    // Fetch the data for the first page
+    await store.actions.source.fetch("/2019/");
+
+    // Fetch the data again (this time returning `date2019PostsPage2`)
+    await store.actions.source.fetch("/2019/", {
+      force: true,
+    });
+
+    // Make sure that api.get() was called for each `source.fetch()`
+    expect(api.get).toHaveBeenCalledTimes(2);
+
+    expect(store.state.source).toMatchSnapshot();
```

Comment:
I'm not 100% if this is the correct approach to take.

After the first fetch the state will contain the first 3 posts with id's `1`, `2` & `3`. Then, after the second fetch I fetch the two posts with id `4` & `5`. 

I'm not sure if the `state.source.post` should only contain the posts `4` & `5` or all the posts from 1 to 5.

To make matters more confusing `state.source.post` contains posts from 1 to 5: https://github.com/frontity/frontity/blob/3770212f44f701f77f521c2ce98009f98d4abd1e/packages/wp-source/src/libraries/handlers/__tests__/__snapshots__/date.tests.ts.snap#L1265-L1300

but `state.source.data` only contains posts 4 and 5:
https://github.com/frontity/frontity/blob/3770212f44f701f77f521c2ce98009f98d4abd1e/packages/wp-source/src/libraries/handlers/__tests__/__snapshots__/date.tests.ts.snap#L1321-L1338

Reply:
It's better to not remove the content once it was fetched. You can overwrite it, but not remove it.

For example, imagine that you navigate to the 2019 archive (`/2019/`), click on the post 1 (`/post-1/attachment-1/`) and while you are reading the content then frontity _updates_ the 2019 archive (e.g. with a link which executes `actions.source.fetch` on hover). The content you are reading would disappear!

Also, even if you haven't visited that post yet, when you visit it Frontity will avoid a WP REST API request because that post is already in the state.

So yeah, it's completely fine to have that data in `state.source.post` and `state.source.data`. :smile:

Reply:
> So yeah, it's completely fine to have that data in state.source.post and state.source.data. 😄

ok, so you're saying that even if I fetch with `force: true` this situation is okay? 

Namely, that after the second fetch (with `force:true`) I have only posts 4 & 5 in `state.source.data` BUT posts 1,2,3,4 & 5 in `state.source.post`